src/OrphanFileHelper.js

```diff
@@ -0,0 +1,82 @@
+const fs = require('fs');
+
+const logger = require('./logging');
+const fileStore = require('./store/file');
+
+/**
+ *
+ */
+class OrphanFileHelper {
+    /**
+     *
+     */
+    constructor({ tempPath, reconnectTimeout, dumpPersister }) {
+        this.tempPath = tempPath;
+        this.reconnectTimeout = reconnectTimeout;
+        this.dumpPersister = dumpPersister;
+    }
+
+    /**
+     * Remove old files from the temp folder.
+     */
+    processOldFiles() {
+        logger.info('[OrphanFileHelper] Waiting for connections to reconnect.');
+        try {
+            if (fs.existsSync(this.tempPath)) {
+                fs.readdirSync(this.tempPath).forEach(fname => {
+                    try {
+                        const filePath = `${this.tempPath}/${fname}`;
+
+                        logger.debug(`[OrphanFileHelper] Trying to process file ${filePath}`);
+                        fs.stat(filePath, (err, stats) => {
```

Comment:
fs.stat is async and it looks like most of the logic here is sync, is this the behavior you're expecting here? i.e. processOldFiles will exist after it finished iterating all the files but  `processIfExpired` will be applied in a async manner after `processOldFiles` has exited.